### PR TITLE
Feat 0613 trace client

### PR DESCRIPTION
**Pull request by wangjinyang**
**What kind of change does this PR introduce?**

**Did you add tests(unit,e2e) for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: packages/shuvi/src/shuvi.ts [hunk 1]
```diff
@@ -18,31 +16,24 @@ export interface ShuviOption {
...
     presets,
     config: restConfig,
     platform: getPlatform(),
-    normalizePlatformConfig,
-    telemetry
+    normalizePlatformConfig
   });
 
+  if (telemetry) {
+    shuvi.trace.setReporter(telemetry.report);
```
[the remark this answers is not in the document]
Comment (liximomo) [reply packages/shuvi/src/shuvi.ts:35]: // client/app.js
// import trace from "@shuvi/service/trace"

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/service/src/core/api.ts [hunk 2]
```diff
@@ -118,11 +109,14 @@ class Api {
     this._customConfig = config || {};
     this._customPresets = presets || [];
     this._customPlugins = plugins || [];
-    this._telemetryImpl = telemetry;
     this._pluginManager = getManager();
     this._pluginManager.clear();
     this._projectBuilder = new ProjectBuilder();
     this._normalizePlatformConfig = normalizePlatformConfig;
+    const shuviTrace = trace('SHUVI_TRACE') as Trace;
```
[the remark this answers is not in the document]
Comment (wangjinyang) [reply packages/service/src/core/api.ts]: done

File: packages/service/src/core/api.ts [hunk 3]
```diff
@@ -145,29 +139,8 @@ class Api {
...
-          if (!this._telemetryImpl) {
-            return Promise.resolve();
-          }
-          return this._telemetryImpl.flush();
-        }
-      };
-    }
-
-    return this._telemetry;
+  get trace() {
```
[the remark this answers is not in the document]
Comment (wangjinyang) [reply packages/service/src/core/api.ts]: done

File: packages/trace/src/index.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // This package 'trace' is a modified version of the Next.js that can be found here:
 // https://github.com/vercel/next.js/tree/canary/packages/next/src/trace
 
-import { trace, flushAllTraces, Span, SpanStatus } from './trace';
-import { SpanId, setGlobal } from './shared';
+import { trace, Span, SpanStatus } from './trace';
+import { SpanId, Reporter, setReporter, getReporter } from './shared';
```
[the remark this answers is not in the document]
Comment (wangjinyang) [reply packages/trace/src/index.ts]: done

**Pull request merged**